homeassistant/components/frigidaire/climate.py

```diff
@@ -0,0 +1,273 @@
+"""ClimateEntity for frigidaire integration."""
+from __future__ import annotations
+
+import logging
+
+import frigidaire
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up frigidaire from a config entry."""
+    client = hass.data[DOMAIN][entry.entry_id]
+
+    def get_entities(username: str, password: str) -> list[frigidaire.Appliance]:
+        return client.get_appliances()
+
+    appliances = await hass.async_add_executor_job(
+        get_entities, entry.data["username"], entry.data["password"]
+    )
+
+    async_add_entities(
+        [FrigidaireClimate(client, appliance) for appliance in appliances],
+        update_before_add=True,
+    )
+
+
+FRIGIDAIRE_TO_HA_UNIT = {
+    frigidaire.Unit.FAHRENHEIT.value: TEMP_FAHRENHEIT,
+    frigidaire.Unit.CELSIUS.value: TEMP_CELSIUS,
+}
+
+FRIGIDAIRE_TO_HA_MODE = {
+    frigidaire.Mode.OFF.value: HVAC_MODE_OFF,
+    frigidaire.Mode.COOL.value: HVAC_MODE_COOL,
+    frigidaire.Mode.FAN.value: HVAC_MODE_FAN_ONLY,
+    frigidaire.Mode.ECO.value: HVAC_MODE_AUTO,
+}
+
+FRIGIDAIRE_TO_HA_FAN_SPEED = {
+    frigidaire.FanSpeed.OFF.value: FAN_OFF,  # when the AC is off
+    frigidaire.FanSpeed.AUTO.value: FAN_AUTO,
+    frigidaire.FanSpeed.LOW.value: FAN_LOW,
+    frigidaire.FanSpeed.MEDIUM.value: FAN_MEDIUM,
+    frigidaire.FanSpeed.HIGH.value: FAN_HIGH,
+}
+
+HA_TO_FRIGIDAIRE_FAN_MODE = {
+    FAN_AUTO: frigidaire.FanSpeed.AUTO,
+    FAN_LOW: frigidaire.FanSpeed.LOW,
+    FAN_MEDIUM: frigidaire.FanSpeed.MEDIUM,
+    FAN_HIGH: frigidaire.FanSpeed.HIGH,
+}
+
+HA_TO_FRIGIDAIRE_HVAC_MODE = {
+    HVAC_MODE_AUTO: frigidaire.Mode.ECO,
```

Comment:
This code has move, but the same comment as before is open.

`HVAC_MODE_AUTO` is used for when on a schedule, learned behavior, or AI.
I don't think eco matches that? It probably be part of presets.